src/controller/python/chip/ChipCoreBluetoothMgr.py

```diff
@@ -230,43 +239,41 @@ def setInputHook(self, hookFunc):
         # set the new hook. readLine will call this periodically as it polls for input.
         pyos_inputhook_ptr.value = cast(self.hookFuncPtr, c_void_p).value
 
-    def shouldLoop(self, should_tuple):
+    def shouldLoop(self, cond):
         """ Used by runLoopUntil to determine whether it should exit the runloop."""
-        result = False
 
-        time_expired = time.time() >= should_tuple[1] + should_tuple[2]
-        if time_expired:
+        if cond.TimeLimitExceeded():
             return False
 
-        if should_tuple[0] == "ready":
+        if cond.op == "ready":
             return not self.ready_condition
-        elif should_tuple[0] == "scan":
+        elif cond.op == "scan":
             for peripheral in self.peripheral_adv_list:
-                if should_tuple[3] and str(peripheral.peripheral._.name) == should_tuple[3]:
+                if cond.arg and str(peripheral.peripheral._.name) == cond.arg:
                     return False
                 devIdInfo = peripheral.getPeripheralDevIdInfo()
-                if devIdInfo and should_tuple[3] and str(devIdInfo.discriminator) == should_tuple[3]:
+                if devIdInfo and cond.arg and str(devIdInfo.discriminator) == cond.arg:
                     return False
-        elif should_tuple[0] == "connect":
+        elif cond.op == "connect":
             return (not self.loop_condition)
-        elif should_tuple[0] == "disconnect":
+        elif cond.op == "disconnect":
             return (not self.loop_condition)
-        elif should_tuple[0] == "send":
+        elif cond.op == "send":
             return (not self.send_condition)
-        elif should_tuple[0] == "subscribe":
+        elif cond.op == "subscribe":
             return (not self.subscribe_condition)
-        elif should_tuple[0] == "unsubscribe":
+        elif cond.op == "unsubscribe":
             return self.subscribe_condition
 
-        return False
+        return True
 
-    def runLoopUntil(self, should_tuple):
+    def runLoopUntil(self, cond):
         """Helper function to drive OSX runloop until an expected event is received or
         the timeout expires."""
```

Comment:
Maybe document that `cond` is a `BgLoopCondition`?

Reply:
Done